### PR TITLE
fix: correct the tool description problem

### PR DESCRIPTION
**Pull request by MrtinoRG**
## Summary by Sourcery

Normalize and enhance tool docstring conventions by introducing explicit ARGS_* and RETURNS_* tags and updating the verbosity router to support these new sections

Enhancements:
- Standardize function docstrings across tools by replacing generic [BRIEF], [DETAILED], [SYNTACTIC], [EXAMPLES] tags with ARGS_* and RETURNS_* sections for arguments and return values
- Extend the router verbosity configuration to recognize and filter new ARGS_ and RETURNS_ docstring sections

Documentation:
- Update ML, Catalyst, and Spectra Elucidation tools to use new ARGS_* and RETURNS_* tags in docstrings

### CODE REVIEW

File: src/corral/router/verbosity.py
Comment (kjappelbaum) [inline src/corral/router/verbosity.py:115]: I'm not sure I would write the code this way.

I would probably use a stricter regex, especially for the keyword sections themselves. For example, I'd use a [lookbehind](https://www.regular-expressions.info/lookaround.html) to ensure that nothing appears before `BRIEF`, `DETAILED`, etc., except perhaps whitespace or newlines.

Then I would separately match the Args and Returns sections.

With this approach, I wouldn't need any if statements. Instead, I would return three dictionaries: one for the keywords, one for the args, and one for the returns. I would then validate that they are consistent with each other.

File: tests/test_verbosity.py [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+FULL_ARG_MP_ID = """[ARGS_BRIEF] Materials Project identifier string. [/ARGS_BRIEF]
+[ARGS_DETAILED] The unique identifier used by Materials Project to catalog materials.
+Should be in the format "mp-XXXXX" where XXXXX is a numerical ID.
+This ID corresponds to a specific material entry in the Materials Project database. [/ARGS_DETAILED]
+[ARGS_SYNTACTIC] "mp-" followed by digits (e.g., "mp-149", "mp-20066") [/ARGS_SYNTACTIC]
+[ARGS_EXAMPLES] "mp-149" (Silicon), "mp-20066" (CO2), "mp-2" (Li) [/ARGS_EXAMPLES]
+"""
+
+
+class TestFilterToolDescription:
```
Comment (kjappelbaum) [inline tests/test_verbosity.py:201]: why did you structure the tests in classes?

File: tests/test_verbosity.py [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+        assert result == "No description available"
+
+        result = VerbosityConfig.filter_tool_description(None, ToolVerbosity.WORKFLOW)
+        assert result == "No description available"
+
+    def test_docstring_without_tags(self):
+        """Test filtering docstring without special tags"""
+        simple_docstring = "This is a simple tool that does something useful."
+        result = VerbosityConfig.filter_tool_description(
+            simple_docstring, ToolVerbosity.BRIEF
```
Comment (kjappelbaum) [inline tests/test_verbosity.py:341]: should we also verify this behavior for other verbosities?

File: tests/test_verbosity.py [hunk 3]
```diff
@@ -0,0 +1,477 @@
...
+
+        # None is returned as-is by the function
+        result = VerbosityConfig.filter_argument_description(
+            None, ToolVerbosity.WORKFLOW
+        )
+        assert result is None or result == ""
+
+    def test_simple_arg_no_tags(self):
+        """Test argument without special tags"""
+        simple_arg = "A simple parameter description"
```
Comment (kjappelbaum) [inline tests/test_verbosity.py:441]: should we also test the case where you have `Args` but not with the suffix for verbosity level?

File: tests/test_verbosity.py [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+
+
+class TestEdgeCases:
+    """Test edge cases and error conditions"""
+
+    def test_malformed_tags(self):
+        """Test handling of malformed tags"""
+        malformed = "[BRIEF] Some text [DETAILED] Missing close tag"
+
+        # Should handle gracefully
```
Comment (kjappelbaum) [inline tests/test_verbosity.py]: should we really handle this gracefully? I feel we would be better off if the parser complained here and we would just throw and exception if the syntax is not what we expect

Comment (kjappelbaum) [reply tests/test_verbosity.py]: I am actually surprised because `_keyword_regex` is relatively harsh (requiring the closing group, which I like)

File: tests/test_verbosity.py (hunk 1 above)
Comment (MrtinoRG) [reply tests/test_verbosity.py:201]: Organization such that the tests for the parsing of the arguments are located and separated from the ones about the entire docstring

File: src/corral/router/verbosity.py
Comment (MrtinoRG) [reply src/corral/router/verbosity.py:115]: I am not sure if I am fully following you.

For the lookbehind, I guess you would look that is the beginning of the line, because in one take into account other lines only for BRIEF applies that there is nothing before.

And then yeah, I guess for the Args and Returns one can have that such that the code is simplified but it is not that much

File: tests/test_verbosity.py (hunk 3 above)
Comment (MrtinoRG) [reply tests/test_verbosity.py:441]: I added that below in the edge cases, with this cases in which Args and Returns do not have keywords

File: tests/test_verbosity.py (hunk 4 above)
Comment (MrtinoRG) [reply tests/test_verbosity.py]: So you would raise an Exception if one of the keywords is missing? But that would not work for the "basic" level, i.e., docstring without keywords.

I guess that perhaps what you have in mind is to raise an Exception if only the opening keyword appears and not the closing. Yeah, I think this is a good suggestion.

Now I implemented this such that when only an opening tag is found, and not the closing counterpart, an Exception is raised 👍

Comment (MrtinoRG) [reply tests/test_verbosity.py]: And I also added some tests for checking that these exceptions are raised :))

**Review by MrtinoRG — COMMENTED** [review]
Thanks for the review @kjappelbaum!

I think I have now addressed all your comments and this should be ready for merging

File: tests/test_verbosity.py [hunk 5]
```diff
@@ -0,0 +1,606 @@
...
+`execute_python_script("processing.py", None, 600, None)`,
+[/SYNTACTICAL]
+
+Args:
+    script_path: [ARGS_BRIEF] Path to the Python script file to execute. [/ARGS_BRIEF]
+                [ARGS_DETAILED] Complete file path to the Python script that should be executed.
+                The script must exist and be readable.
+                The path can be relative to the current working directory or absolute.
+                The script should be a valid Python file with appropriate shebang or run using the Python interpreter. [/ARGS_DETAILED]
+                [ARGS_SYNTACTIC] "Valid file path to Python script" [/ARGS_SYNTACTIC]
```
Comment (n0w0f) [inline tests/test_verbosity.py]:
```suggestion
                [ARGS_SYNTACTICAL] "Valid file path to Python script" [/ARGS_SYNTACTICAL]
```

Shouldnt this be ARGS_SYNTACTICAL ?

**Review by n0w0f — APPROVED** [review]

File: tests/test_verbosity.py (hunk 5 above)
Comment (MrtinoRG) [reply tests/test_verbosity.py]: Yes, you were right 🙃
